src/scripts/location-select.js

```diff
@@ -0,0 +1,137 @@
+ * @returns {Boolean}
+ */
+function goodPrecip(shortForecast) {
+    const exclude = ['rain', 'shower'].map((word) => `(?!.*\\b${word}\\b)`).join('');
+    const rgx = new RegExp(`^${exclude}.*$`, 'i');
+    return rgx.test(shortForecast);
+}
+
+/**
+ * Determines if the humidity is adequate.
+ * 
+ * @param {Number} - The relativeHumidity value from the forecast. 
+ * @returns {Boolean}
+ */
+function goodHumidity(relativeHumidity) {
+    return relativeHumidity.value <= humidityThreshold;
+}
+
+/**
+ * Determines if the windspeed is adequate.
+ * 
+ * @param {String} windspeed - The windspeed from the forecast. 
+ * @returns {Boolean}
+ */
+function goodWindspeed(windspeed) {
+    return parseInt(windspeed.replace(/\D+/, ''), 10) < windSpeedThreshold;
+}
+
+/**
+ * Determines which of the given forecasts are adequate using Array.reduce().
+ * 
+ * @param {Array<Object>} acc - Accumulation of good forecasts.
+ * @param {Object} forecast - A forecast to check.
+ * @returns {Array<Object>} - The final accumulation of good forecasts.
+ */
+function goodForecasts(acc, forecast) {
+    const {
+        temperature,
+        startTime,
+        shortForecast,
+        relativeHumidity,
+        windSpeed
+    } = forecast;
+
+    if (
+        !goodHours(startTime)
+        || !goodTemperature(temperature)
+        || !goodHumidity(relativeHumidity)
+        || !goodPrecip(shortForecast)
+        || !goodWindspeed(windSpeed)
+    ) return acc;
+    return acc.concat(forecast);
+}
+
+// On load, fetch cities.json for input field, render as options.
+fetch('/cities.json').then((res) => res.json()).then((cities) => {
+    $cities.innerHTML = cities.map((entry) => {
+        return `<option>${formatLabel(entry)}</option>`;
+    }).join('');
+
+    /**
+     * Attempts to trigger a series of requests to fetch weather and map data.
+     * 
+     * This function sits inside of the earlier fetch results to hold the `cities` data.
+     * 
+     * @param {String} value - Value within the input field. 
+     */
+    function onChange(value) {
+        const city = cities.find((entry) => formatLabel(entry) === value);
+        if (!city) return;
+        // Event is read by <Map/> component.
+        window.dispatchEvent(new CustomEvent('city', { detail: formatLabel(city) }));
+        const url = new URL(`/points/${city.latitude},${city.longitude}`, WEATHER_GOV_BASE);
+        fetch(url.toString())
+            .then((res) => res.json())
+            .then(({ properties }) => fetch(properties.forecastHourly))
+            .then((res) => res.json())
+            .then(({ properties }) => {
+                const forecasts = properties.periods.reduce(goodForecasts, []);
+                // Event is read by <Results/> and <Calendar/> components.
+                window.dispatchEvent(new CustomEvent('forecasts', { detail: forecasts }));
+            });
+    }
+
+    $input?.addEventListener('change', ({ target }) => onChange($input.value));
```

Comment:
not sure if this is the right place to leave a comment haha:

Can you truncate the selected city text again in the text field? So when `New York, New York` is selected, the text field only displays `New York`.

Reply:
This is trickier than it should be. The `<input/>` element has the listener for `onChange`, which listens for when the input updates and tries to do a [lookup against the `cities.json` data with the `value` from the `<input/>`](https://github.com/stanchanpsu/whencanipickle/pull/22/files#diff-aa50900740b818cdb44d2cfa782878d177d354c024ffea336590d6fcfd70a626R119). If the text in the input is only "New York", the lookup would fail since it doesn't match anything directly with the data.

This is the disappointing behavior of the `<datalist/>` element I put in there to mimic the earlier experience. The experience we want is for the `<input/>` to filter the list (which it does) but for selection in the list to trigger the `onChange`, not for the selection to update the input.

I just found [something](https://stackoverflow.com/a/71781912/3928045) that _might_ help. I'm hoping I can prevent the replacement from happening but still capture the value. Otherwise, I might need to write additional JS for behaviors.

Reply:
First: this is totally not a blocker and we can iterate on it (I can do it) - we can merge first, just let me know.

Is the lookup logic using both the `city` and the `state` for lookups in `cities.json.js`? It's a bit unclear to me how the lookup is performed since the data in the json is itself split and not a concatenated string. Would augmenting the data help at all (creating new fields)?

Reply:
Ok did the easy thing. After the selection occurs, I replace the `value` with the `location.city`.

Reply:
> Is the lookup logic using both the `city` and the `state` for lookups in `cities.json.js`? It's a bit unclear to me how the lookup is performed since the data in the json is itself split and not a concatenated string. Would augmenting the data help at all (creating new fields)?

There's a lot to be desired in the `<datalist/>`. What I'd _really_ like to do is to read the `value` from the selected option in the `<datalist/>` in order to perfectly identify the user's selection. There's two problems with this:

- Writing the `value` in the `<option/>` will _display the value_, at least on my Chromebook. This is undesirable as we only want to show the content or the `label`.
- I can't listen for events on the `<datalist/>`, only the `<input/>` and that won't provide a `value` of the selected option, only the text.

Because of both of those qualities, I need to do a look up using the exact text used as the display within the `<datalist/>`. I do the city+state concat in [this function](https://github.com/stanchanpsu/whencanipickle/pull/22/files#diff-aa50900740b818cdb44d2cfa782878d177d354c024ffea336590d6fcfd70a626R18) which creates the options and checks for the comparison in the lookup.

Reply:
We can replace this component with something more appropriate in the future.